lib/nerves_network/wifi_manager.ex

```diff
@@ -388,23 +399,67 @@ defmodule Nerves.Network.WiFiManager do
         :nameservers
       ])
 
-    case Nerves.WpaSupplicant.set_network(pid, wpa_supplicant_settings) do
-      :ok ->
-        :ok
+    networks = parse_settings(wpa_supplicant_settings)
 
-      error ->
-        Logger.info(
-          "WiFiManager(#{state.ifname}, #{state.context}) wpa_supplicant set_network error: #{
-            inspect(error)
-          }"
-        )
+    Nerves.WpaSupplicant.remove_all_networks(pid)
 
-        notify(Nerves.WpaSupplicant, state.ifname, error, %{ifname: state.ifname})
+    for network <- networks do
+      case Nerves.WpaSupplicant.add_network(pid, network) do
+        {:ok, _} ->
+          :ok
+
+        error ->
+          Logger.info(
```

Comment:
I wonder if this should be `warn` or `error` instead? I'm not sure what kinds of errors would appear here.

Reply:
I wondered this also, but kept it as `info` because that's what it used to be. I think i'll change it to error because this one can be tricky to find.

it happens when you make a typo or some other validation issued. example: `psk: "short"` that psk isn't long enough (i think shortest is 8 chars). wpa_supplicatnt wont say _why_ it is wrong, but it will tell you it failed to use that key.

Reply:
Yeah my thinking was that any time things didn't work, we want to surface that to the user in a way they'll hopefully know what to do about it. I know we have a lot of useless debug stuff, but this seems like a truly useful error message to want to show people if they configured network settings and it didn't work (for some reason other than the requested network just isn't available, which I guess would be a `info`-level thing).